src/twister2/twister_config.py

```diff
@@ -153,6 +150,31 @@ def verify_platforms_existence(self, platform_names_to_verify: list[str]):
                 raise TwisterConfigurationException(msg)
 
 
+def _get_hardware_map_list(config: pytest.Config) -> list[HardwareMap]:
+    """Process input parameters and return list of hardware maps"""
+    hardware_map_list: list[HardwareMap] = []
+    if config.option.hardware_map:
+        hardware_map_list = HardwareMap.read_from_file(filename=config.option.hardware_map)
+    elif config.option.device_serial:
+        hardware_map_list = [
+            HardwareMap(
+                connected=True,
+                platform=config.option.platform[0],
```

Comment:
I'm wondering if shouldn't we verify if users provide more than one platform, when it use `--device-serial` or `--device-serial-pty`. For example, for this case:
```
pytest samples/hello_world --device-testing  --platform=nrf52dk_nrf52832 --platform=nrf52840dk_nrf52840 --device-serial=/dev/ttyACM0
```
Twister should rise an error at the beginning, because it doesn't have sense, to pass one `--device-serial` for two platforms.

Moreover you can verify if user provide any platform, before taking first one from the list:
```python
if len(config.option.platform) > 0:
    platform=config.option.platform[0]
```

Reply:
Some validation is done in `validate_options` method in `plugin.py:L296`
```
pytest samples/hello_world --device-testing  --platform=nrf52dk_nrf52832 --platform=nrf52840dk_nrf52840 --device-serial=/dev/ttyACM0
Exit: When `--device-testing` is used with `--device-serial` or `--device-serial-pty`, only one platform is allowed.
```
I agree with checking if any platform is provided (now we have an exception, because we call len mentod on NoneType). I will add that later, but not in `validate_options` because a platform can be provided in a hardware_map file

Reply:
Added validation in `validate_options`:
```
pytest samples/hello_world --device-testing  --device-serial=/dev/ttyACM0
Exit: When `--device-testing` is used with `--device-serial` or `--device-serial-pty`, a platform must be provided.
```